scene/theme/theme_owner.h

```diff
@@ -0,0 +1,75 @@
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef THEME_OWNER_H
+#define THEME_OWNER_H
+
+#include "core/object/object.h"
+#include "scene/resources/theme.h"
+
+class Control;
+class Node;
+class Window;
+
+class ThemeOwner : public Object {
+	Control *owner_control = nullptr;
+	Window *owner_window = nullptr;
+
+	Node *_get_next_owner_node(Node *p_from_node) const;
+
+public:
+	// Theme owner node.
+
+	void set_owner_node(Node *p_node);
+	Node *get_owner_node() const;
+	bool has_owner_node() const;
+
+	// Theme propagation.
+
+	void assign_theme_on_parented(Node *p_for_node);
+	void clear_theme_on_unparented(Node *p_for_node);
+	void propagate_theme_changed(Node *p_to_node, Node *p_owner_node, bool p_notify, bool p_assign);
+
+	// Theme lookup.
+
+	void get_theme_type_dependencies(const Node *p_for_node, const StringName &p_theme_type, List<StringName> *r_list) const;
+
+	Variant get_theme_item_in_types(Theme::DataType p_data_type, const StringName &p_name, List<StringName> p_theme_types);
+	bool has_theme_item_in_types(Theme::DataType p_data_type, const StringName &p_name, List<StringName> p_theme_types);
+
+	float get_theme_default_base_scale();
+	Ref<Font> get_theme_default_font();
+	int get_theme_default_font_size();
+
+	ThemeOwner(){};
+	~ThemeOwner(){};
```

Comment:
```suggestion
	ThemeOwner() {}
	~ThemeOwner() {}
```
The semicolon is not needed since there's a scope, and clang-format makes a mess of this syntax.

Reply:
Indeed it does. Removing them makes it behave?

Reply:
Yes, the format I suggested is what `clang-format` would use without the semicolons.